pulsar-transaction/coordinator/src/main/java/org/apache/pulsar/transaction/coordinator/impl/TxnLogBufferedWriter.java

```diff
@@ -272,23 +290,43 @@ private void doTrigFlush(boolean force, boolean byScheduleThreads){
                 return;
             }
             if (force) {
+                if (metricsStats != null) {
```

Comment:
We can add a default TxnLogBufferedWriterMetricsStatsDisabled implementation to avoid many null checks.

Reply:
> We can add a default TxnLogBufferedWriterMetricsStatsDisabled implementation to avoid many null checks.

Good Idea. In the complete design, we should have two implementations like UML blow, one for enabling the batch feature, and another for disabled:

![uml](https://user-images.githubusercontent.com/25195800/182418488-6469a38f-a96c-44e9-8ee6-01273b58b0cd.jpeg)


Sorry, I should have added some comments here.
I didn't want to support disabled metrics stat in the design, even if the Txn Buffered Writer disabled the batch feature. This constructor without `metricsStats` and "null checks" will be removed in the next PR. This is only compatible to split the PR, making each PR has less code